exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/join/HashJoinMemoryCalculatorImpl.java

```diff
@@ -391,7 +391,7 @@ private void calculateMemoryUsage()
         // probe batch we sniffed.
         // TODO when batch sizing project is complete we won't have to sniff probe batches since
         // they will have a well defined size.
-        reservedMemory = incompletePartitionsBatchSizes + maxBuildBatchSize + maxProbeBatchSize;
+        reservedMemory = incompletePartitionsBatchSizes + maxBuildBatchSize + probeSizePredictor.getBatchSize();
```

Comment:
I'm not sure. I think we need to get some clarification for the desired behavior of operators. If the Drill engine is operating under the assumption that ownership of VectorContainers passes to a downstream operator after a call to next, then this is a bug that should be fixed.

I will ask a question on the dev list about this.

Reply:
@sohami explained that it is up to the operator whether or not to transfer ownership of the incoming vector batch and described the two primary use cases for transferring or not transferring ownership. Looking at the HashJoin code we never transfer ownership of the incoming probe batch and copy the probe data directly to the outgoing batch. This explains why you never got an OOM. 

So I think you are right, we should not be reserving space for the incoming probe batch, since it is never owned by HashJoin's allocator. Thanks for catching this @Ben-Zvi I will update the PR.

Reply:
@Ben-Zvi thought about this a bit more and there is a complication with spilling. If we are doing the join on a spilled partition the HashJoin operator will actually own the probe batch. So during the first cycle we don't want to account for the probe batch because the upstream operator will own it, but during the other spill cycles we do want to account for them because we will own them.